base/abstractarray.jl

```diff
@@ -1829,19 +1829,24 @@ function mapslices(f, A::AbstractArray, dims::AbstractVector)
     R[ridx...] = r1
 
     nidx = length(otherdims)
+    indexes = Iterators.drop(CartesianRange(itershape), 1)
+    inner_mapslices!(safe_for_reuse, indexes, nidx, idx, otherdims, ridx, Aslice, A, f, R)
+end
+
+@noinline function inner_mapslices!(safe_for_reuse, indexes, nidx, idx, otherdims, ridx, Aslice, A, f, R)
     if safe_for_reuse
```

Comment:
It looks like `safe_for_reuse` could now more idiomatically be expressed with dispatch (and two `inner_mapslices!` methods).  Of course the performance impact here is a drop in the ocean… but I think it's a nicer way to express the two algorithms.